src/menu-renderer/index.ts

```diff
@@ -137,13 +143,22 @@ Promise.all([
   // Report unhover events to the main process.
   menu.on('unhover', (path) => window.menuAPI.unhoverItem(path));
 
-  // Hide the menu when the user presses escape.
-  document.body.addEventListener('keydown', (ev) => {
+  document.body.addEventListener('keydown', async (ev) => {
+    // Hide the menu when the user presses escape.
     if (ev.key === 'Escape') {
       menu.hide();
       settingsButton.hide();
       window.menuAPI.cancelSelection();
     }
+
+    if ((await window.commonAPI.appSettings.get()).useDefaultOsShowSettingsHotkey) {
+      // Show the settings window if 'cmd + ,' hotkey is pressed on macOS
+      // or 'ctrl + ,' is pressed on non macOS systems.
+      const keyIsComma = ev.key === ',';
+      if ((ev.metaKey && keyIsComma && cIsMac) || (ev.ctrlKey && keyIsComma && !cIsMac)) {
```

Comment:
It just occurred to me that it may be better to switch the if's to avoid the asynchronous IPC call on every key event. I guess that would be better, or what do you think?

```suggestion
    // Show the settings window if 'cmd + ,' hotkey is pressed on macOS
    // or 'ctrl + ,' is pressed on non macOS systems.
    const keyIsComma = ev.key === ',';
    if ((ev.metaKey && keyIsComma && cIsMac) || (ev.ctrlKey && keyIsComma && !cIsMac)) {
      if ((await window.commonAPI.appSettings.get()).useDefaultOsShowSettingsHotkey) {
```

Reply:
I 100% agree. I had it this way previously before the first commit, I don't know why I moved it around haha.